extensions/test/CognitoAuthentication.IntegrationTests/BaseAuthenticationTestClass.cs

```diff
@@ -42,8 +42,11 @@ public BaseAuthenticationTestClass()
             UserPoolPolicyType passwordPolicy = new UserPoolPolicyType();
             List<SchemaAttributeType> requiredAttributes = new List<SchemaAttributeType>();
             List<string> verifiedAttributes = new List<string>();
-
+#if  XAMARIN_TESTS
```

Comment:
It might be a better idea to abstract this platform differences in to different folders.  See this [folder structure](https://github.com/aws/aws-sdk-net/tree/master/sdk/src/Services/EC2/Custom) and csproj [filtering](https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Services/EC2/AWSSDK.EC2.CoreCLR.csproj#L36).

We may want to create partial classes that abstract this platform difference:

in `/_bcl/BaseAuthenticationTestClass.cs`:
```
public partial class BaseAuthenticationTestClass{
    AmazonCognitoIdentityProviderClient GetAmazonCognitoIdentityProviderClient(...){
         return new AmazonCognitoIdentityProviderClient(Settings.storedSettings.AccessKeyId,Settings.storedSettings.SecretAccessKey, Settings.RegionEndpoint);
    }
}
```

in `/_xamarin/BaseAuthenticationTestClass.cs`:
```
public partial class BaseAuthenticationTestClass{
    AmazonCognitoIdentityProviderClient GetAmazonCognitoIdentityProviderClient(...){
         return new AmazonCognitoIdentityProviderClient();
    }
}
```

Reply:
Do we need to implement this this time?